### ISSUE TITLE
Support multiple loss functions

### ISSUE THREAD

**Issue by mathpluscode** [DeepRegNet/DeepReg#526]
## Subject of the feature

What are you trying to do and how would you want to do it differently?

I'd like to use multiple loss functions together, e.g. dice loss + cross-entropy.
But it is not possible for the moment I guess.

If the feature request is approved, would you be willing to submit a PR? _(Help can be
provided if you need assistance submitting a PR)_

Yes

**Comment by NMontanaBrown** [comment]
I have had a vague idea for this now that I've worked on the config docs. @acasamitjana are you working on this right now?

**Comment by acasamitjana** [comment]
No, I haven't started yet, so go ahead. I'm happy to help if needed.

**Comment by mathpluscode** [comment]
This issue is doable now with the registry :) Will any of you be still interested and available? @acasamitjana @NMontanaBrown

**Comment by NMontanaBrown** [comment]
I'd close this if the other issue #512 is open and is going to use registry...?

> This issue is doable now with the registry :) Will any of you be still interested and available? @acasamitjana @NMontanaBrown

**Comment by mathpluscode** [comment]
> I'd close this if the other issue #512 is open and is going to use registry...?
> 
> 
> 
> > This issue is doable now with the registry :) Will any of you be still interested and available? @acasamitjana @NMontanaBrown
> 
> 
> 
> 

@NMontanaBrown I thought the multiple losses here mean:
You can apply two different losses to images at the same time, like LNCC loss and SSD ;)

**Comment by NMontanaBrown** [comment]
Ah yes, I misread your comment as double not doable. I will not be able to work for this for the next couple of weeks actively, as I will be busy with some other things. So if they want to, someone else should pick this up, but otherwise I will return to it after a couple of weeks :) @mathpluscode

**Comment by mathpluscode** [comment]
> Ah yes, I misread your comment as double not doable. I will not be able to work for this for the next couple of weeks actively, as I will be busy with some other things. So if they want to, someone else should pick this up, but otherwise I will return to it after a couple of weeks :) @mathpluscode 

Sure, I might handle it myself. It's a small change :)

**Comment by acasamitjana** [comment]
It allows multiple loss functions by specifying a list under each loss type: "image", "label", "regularization".
The way to do so is using a dashed sign (-). A config file example is:

```
loss:
  "image"
      - "name": "ssd"
         "weight": 0.5
      - "name": "lncc"
         "weight": 0.5
  "label"
       "name": "cross-entropy"
       "weight": 1.0
  "regularization":
        "name": "bending"
        "weight": 0.1
...
```
